# Smith is in jail and has 3 dollars; he can get out on bail if he has 8 dollars

# A guard agrees to make a series of bets with him

# If Smith bets $A$ dollars, he wins $A$ dollars with probability 0.4 and wins $A$ dollars with probability 0.6

# Find the probability that he wins 8 dollars before losing all his money if:

## a) he bets 1 dollar each time (timid strategy)

## b) he bets, each time, as much as possible but not more than necessary to bring his fortune up to 8 dollars (bold strategy)

## c) Which strategy gives Smith the better chance of getting out of jail?

____

# a) Timid Strategy

## There are 9 states each representing the number of dollars Smith has won: [1,2,3,4,5,6,7,8,0]

## When he loses all his money, he cannot play anymore

## When he has won 8 dollars, he stops playing

## That means that both 0 and 8 are absorbing states

## We can represent the process with the following transition matrix

![](images/13.PNG)

# $\implies Q = \begin{pmatrix}0 & 0.4 & 0 & 0 & 0 & 0 & 0\\ 0.6 & 0 & 0.4 & 0 & 0 & 0 & 0\\ 0 & 0.6 & 0 & 0.4 & 0 & 0 & 0\\ 0 & 0 & 0.6 & 0 & 0.4 & 0 & 0\\ 0 & 0 & 0 & 0.6 & 0 & 0.4 & 0\\ 0 & 0 & 0 & 0 & 0.6 & 0 & 0.4\\ 0 & 0 & 0 & 0 & 0 & 0.6 & 0\end{pmatrix}$

# $\implies (I-Q) = \begin{pmatrix}1 & -0.4 & 0 & 0 & 0 & 0 & 0\\ -0.6 & 1 & -0.4 & 0 & 0 & 0 & 0\\ 0 & -0.6 & 1 & -0.4 & 0 & 0 & 0\\ 0 & 0 & -0.6 & 1 & -0.4 & 0 & 0\\ 0 & 0 & 0 & -0.6 & 1 & -0.4 & 0\\ 0 & 0 & 0 & 0 & -0.6 & 1 & -0.4\\ 0 & 0 & 0 & 0 & 0 & -0.6 & 1\end{pmatrix}$

**Using numpy to solve for the inverse**

In [2]:
import numpy as np
import pandas as pd

In [3]:
matrix = np.array([[1, -0.4, 0, 0, 0, 0, 0],
                  [-0.6, 1, -0.4, 0, 0, 0, 0],
                  [0, -0.6, 1, -0.4, 0, 0, 0],
                  [0, 0, -0.6, 1, -0.4, 0, 0],
                  [0, 0, 0, -0.6, 1, -0.4, 0],
                   [0, 0, 0, 0, -0.6, 1, -0.4],
                  [0, 0, 0, 0, 0, -0.6, 1]])

In [4]:
N = np.linalg.inv(matrix)

# We know that to solve for the absorption probabilities, we must calculate $NR$

# $R = \begin{pmatrix}0 & 0.6\\ 0 & 0\\0 & 0\\0 & 0\\0 & 0\\0 & 0\\0.4 & 0\end{pmatrix}$

In [5]:
R = np.array([[0, 0.6],
             [0,0],
             [0,0],
             [0,0],
             [0,0],
             [0,0],
             [0.4,0]])

In [6]:
NR = np.matmul(N, R)

In [7]:
NR

array([[ 0.02030135,  0.97969865],
       [ 0.05075337,  0.94924663],
       [ 0.0964314 ,  0.9035686 ],
       [ 0.16494845,  0.83505155],
       [ 0.26772403,  0.73227597],
       [ 0.42188739,  0.57811261],
       [ 0.65313243,  0.34686757]])

# Since we know we start with 3 dollars, we only care about the third row

# $[0.0964314, 0.9035686]$

# This means that the probability that Smith gets to 8 dollars is only 0.0964

______

# b)

# We repeat the same analysis for the bold strategy

- Starting in state 3, we can go to state 6 (with probability 0.4) or state 0 (with probability 0.6)
- From state 6, we can go to state 8 (with probability 0.4) or state 4 (with probability 0.4)
- From state 4, we can go to state 8 (with probability 0.4) or state 0 (with probability 0.4)

- That means it's impossible to go to any other states

**This means we have a matrix $P$ given by:**

# $P =\begin{pmatrix}0 & 0.4 & 0 & 0 & 0.6\\ 0.6 & 0 & 0 & 0.4 & 0\\ 0 & 0 & 0 & 0.4 & 0.6\\ 0 & 0 & 0 & 1 & 0\\ 0 & 0 & 0 & 0 & 1\end{pmatrix}$

**Where the rows/columns represent states 3, 6, 5, 8, and 0 respectively**

# $\implies Q = \begin{pmatrix}0 & 0.4 & 0\\ 0.6 & 0 & 0\\ 0 & 0 & 0\end{pmatrix}$

# $\implies (I-Q) = \begin{pmatrix}1 & -0.4 & 0\\ -0.6 & 1 & 0\\ 0 & 0 & 1\end{pmatrix}$

In [8]:
matrix = np.array([[1, -0.4, 0], [-0.6, 1, 0], [0, 0, 1]])

In [9]:
N = np.linalg.inv(matrix)

In [10]:
N

array([[ 1.31578947,  0.52631579,  0.        ],
       [ 0.78947368,  1.31578947,  0.        ],
       [ 0.        ,  0.        ,  1.        ]])

# $\implies N = \begin{pmatrix}1.31578947 & 0.52631579 & 0\\ 0.78947368 & 1.31578947 & 0\\ 0 & 0 & 1\end{pmatrix}$

# $R = \begin{pmatrix}0 & 0.6\\0.4 & 0\\0.4 & 0.6\end{pmatrix}$

In [11]:
R = np.array([[0,0.6],[0.4,0],[0.4,0.6]])
NR = np.matmul(N, R)

In [12]:
NR

array([[ 0.21052632,  0.78947368],
       [ 0.52631579,  0.47368421],
       [ 0.4       ,  0.6       ]])

# Since we only care about the probabilities for starting in state 3, we only consider the first row

# $[0.21052632,  0.78947368]$

## This means that the probability of winning is 0.21 $\implies$ this is much better than in the timid strategy